pp-mock/handlers/state/conversion_v1.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2022-2025, Chain4Travel AG. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package state
+
+import typesv1 "buf.build/gen/go/chain4travel/camino-messenger-protocol/protocolbuffers/go/cmp/types/v1"
+
+func ExtractCurrencyV1FromUnifiedPrice(uPrice *UnifiedPrice) (*typesv1.Currency, bool) {
+	currency := &typesv1.Currency{}
+
+	switch {
+	case uPrice.IsNative:
+		currency.Currency = &typesv1.Currency_NativeToken{}
+	case uPrice.IsoCurrencyEnum != 0:
+		currency.Currency = &typesv1.Currency_IsoCurrency{
+			IsoCurrency: typesv1.IsoCurrency(uPrice.IsoCurrencyEnum),
+		}
+	case uPrice.TokenContractAddress != "":
+		currency.Currency = &typesv1.Currency_TokenCurrency{
+			TokenCurrency: &typesv1.TokenCurrency{
+				ContractAddress: uPrice.TokenContractAddress,
+			},
+		}
+	default:
+		return nil, false
+	}
+	return currency, true
+}
+
+func UnifiedPriceToProtoPriceV1(uPrice *UnifiedPrice) (*typesv1.Price, bool) {
```

Comment:
```suggestion
func UnifiedPriceToPriceV1(uPrice *UnifiedPrice) (*typesv1.Price, bool) {
```
Imo, thats enough, we don't have versioned prices anywhere else. Same for similar places including reverse casting
You can also do
```go
func (p *UnifiedPrice) ToPriceV1() (*typesv1.Price, bool) {
```